ibis/backends/sql/compilers/base.py

```diff
@@ -1587,22 +1603,25 @@ def add_query_to_expr(self, *, name: str, table: ir.Table, query: str) -> str:
         dialect = self.dialect
 
         compiled_ibis_expr = self.to_sqlglot(table)
+        compiled_query = sg.parse_one(query, read=dialect)
+
+        ctes = [
+            *compiled_ibis_expr.ctes,
+            sg.exp.CTE(
+                alias=sg.to_identifier(name, quoted=self.quoted),
+                this=compiled_ibis_expr,
+            ),
```

Comment:
It looks like this is the main change in this part of the code.

Is the issue that in the previous implementation `compiled_query` might reference `compiled_ibis_expr` or its CTEs, which would break topological ordering of CTE definition?

Reply:
Hm, not sure I'm following why (1) is a problem.

It seems like (1) is a scenario that is tough to get into. I realize that your tests are probably covering this, so just bear with me.

It sounds like maybe you're suggesting that it's possible for `with_select_statement` to **already** be referencing `alias` before its `with_` method is called, which seems super weird.

Do you have an example of that scenario, or can you point me to which covers this case?

Reply:
I think the var names I picked above are a bit confusing, so just to make sure we have the right ordering in our head let's use `test_embedded_cte_with_alias_simple` as the example: `expr` is the user Ibis expression with an alias, and the user calls `.sql(<with statement>)` on it, where the with statement contains a reference to the alias.

Without this PR, Ibis will call `parsed_user_query.with_(parsed_ibis_expr)`, which is how you end up with the weird case where `with_select_statement` already references the alias.